api_spike.rb

```diff
@@ -0,0 +1,88 @@
+require "faraday"
+require "json"
+
+ingredients = ["apples", "cinnamon"]
+params = { ingredients: ingredients.join(','),
+           time_restraint: 60,
+           diet_restraint: "vegetarian",
+           type_restraint: "dessert" }
+
+def get_recipes(params)
+  request = connection(params).get("/recipes/complexSearch")
+  JSON.parse(request.body, symbolize_names: true)
+end
+
+def connection(params)
+  Faraday.new(url: "https://api.spoonacular.com/",
+              params: { apiKey: ENV['SPOONACULAR_KEY],
+                        addRecipeInformation: true,
+                        includeIngredients: params[:ingredients],
+                        fillIngredients: true,
+                        ignorePantry: true,
+                        maxReadyTime: params[:time_restraint],
+                        diet: params[:diet_restraint],
+                        type: params[:type_restraint],
+                        minCalories: 0,
+                        minFat: 0,
+                        minProtein: 0,
+                        minCarbs: 0,
+                        minSugar: 0,
+                        minSodium: 0,
+                        minCholesterol: 0,
+                        number: 2,
+                        sort: 'random'})
```

Comment:
Do these queries always need to include every parameter?  Do they not work correctly unless everything is filled out?

Reply:
I think so.. I don't know for sure about the minSodium or number values, but I know the ones where we're passing in params it gets an API serverside error. Thankfully `.compact` is an easy way to remove values from a hash if their values are nil or empty.